icu4c/source/i18n/chnsecal.cpp

```diff
@@ -1177,10 +1177,14 @@ int32_t ChineseCalendar::internalGetMonth(int32_t defaultValue, UErrorCode& stat
     if (U_FAILURE(status)) {
         return 0;
     }
-    if (resolveFields(kMonthPrecedence) == UCAL_MONTH) {
-        return internalGet(UCAL_MONTH, defaultValue);
+    switch (resolveFields(kMonthPrecedence)) {
+        case UCAL_MONTH:
+            return internalGet(UCAL_MONTH);
+        case UCAL_ORDINAL_MONTH:
+            return internalGetMonth(status);
```

Comment:
why are you no longer passing the default values into these functions?
circular calls?
is that a problem for both of these?
should they both call internalGet() instead of internalGetMonth()?

Reply:
if we already know the UCAL_MONTH is reolved to, then it must not be unset. In that case, there are no need to pass in the defaultValue since we already know we won't use the default. 
For the case of UCAL_ORDINAL_MONTH, we know there are value stored in the UCAL_ORDINAL_MONTH field, so we will use that value to calculate the month, therefore no need to pass in the default value. 
The default value is used in the condition that neither the field UCAL_MONTH is set nor UCAL_ORDINAL_MONTH is set.

Reply:
Ok, thanks. These things might benefit from comments, but maybe they are obvious to someone who knows this code.